saasherder/changelog.py

```diff
@@ -148,40 +194,54 @@ def generate(self, context, old, new):
         # Where am I right now?
         try:
             # This will fail for detached HEAD
-            branch = self.run("git symbolic-ref --short HEAD").strip()
+            branch = run("git symbolic-ref --short HEAD").strip()
             logger.info("On branch {} before generating changelog"
                         .format(branch))
 
         except Exception:
-            branch = self.run("git rev-parse --short HEAD").strip()
+            branch = run("git rev-parse --short HEAD").strip()
             logger.info("HEAD at {} before generating changelog"
                         .format(branch))
 
+        # Resolve commit if a date was passed in 'old'
+        try:
+            dt_old = parse(old)
+            old = run("git rev-list -1 --before='{}' {}".format(dt_old, branch)).strip()
+        except:
+            pass
```

Comment:
1. Except a more specific class of Error here.
2. Should you `exit(1)` instead of pass?

Reply:
I believe this is correct, `old` and `new` can be one of three things:

- a branch, for example: master
- a commit (short or long), for example: c589a14
- a date, for example: 2018-02-21 10:00

This piece of code will try to detect if it's a date (by doing dateutils.parse). If that parsing fails it shouldn't do anything and keep the original value as it's a commit or a branch. However if the parsing succeeds, then we need to find the commit associated to that time.

However I will refactor this a bit and add some comments, if it wasn't clear when reading the code it means that I need to fix it :)

Reply:
You are right about excepting the specific class of Error, I hadn't read that. Will do.